Cpp2IL.Core/ProcessingLayers/CallAnalysisProcessingLayer.cs

```diff
@@ -237,12 +239,51 @@ private static void AddAttribute((InjectedMethodAnalysisContext, InjectedFieldAn
             parametersField = null;
         }
 
+        (FieldAnalysisContext, object)? returnTypeField;
+        TypeAnalysisContext? returnType;
+        if (targetMethod is NativeMethodAnalysisContext)
+        {
+            returnType = null;//Native methods don't have identifiable return types.
+        }
+        else if (targetMethod.InjectedReturnType is not null)
+        {
+            returnType = targetMethod.InjectedReturnType;
+        }
+        else if (concreteMethod is not null && (concreteMethod.BaseMethodContext.Definition is not null || concreteMethod.BaseMethodContext.InjectedReturnType is not null))
```

Comment:
Code style wise I'd prefer this:
```suggestion
        else if (concreteMethod is { BaseMethodContext: { Definition: not null } or { InjectedReturnType: not null } })
```